lxd/device/disk.go

```diff
@@ -549,9 +556,63 @@ func (d *disk) createDevice() (string, error) {
 	isReadOnly := shared.IsTrue(d.config["readonly"])
 	isRecursive := shared.IsTrue(d.config["recursive"])
 
+	mntOptions := d.config["raw.mount.options"]
+	fsName := ""
+
 	isFile := false
 	if d.config["pool"] == "" {
 		isFile = !shared.IsDir(srcPath) && !IsBlockdev(srcPath)
+		if strings.HasPrefix(d.config["source"], "cephfs:") {
+			isFile = false
+			//filesystem mount
+			userName := d.config["ceph.user_name"]
+			if userName == "" {
+				userName = "admin"
+			}
+			clusterName := d.config["ceph.cluster_name"]
+			if clusterName == "" {
+				clusterName = "ceph"
+			}
+			//ap
+			mntSrcPath, fsOptions, fsErr := diskCephfsOptions(clusterName, userName, fsName)
+			if fsErr != nil {
+				return "", fsErr
+			}
+			mntOptions += fsOptions
```

Comment:
This needs proper joining if mntOptions isn't empty.

Reply:
what can we expect mntOptions to have? like a string delimited by commas so it should be joined with a comma?

Reply:
I've pushed a version with that assumption, but I am not fully sure

Reply:
yep, should be joined by commas